core/src/main/java/io/questdb/griffin/engine/groupby/YearTimestampMicrosSampler.java

```diff
@@ -56,9 +56,9 @@ public long nextTimestamp(long timestamp) {
     }
 
     @Override
-    public long nextTimestamp(long timestamp, int numSteps) {
+    public long nextTimestamp(long timestamp, long numSteps) {
         try {
-            return addYears(timestamp, Math.multiplyExact(numSteps, stepYears));
+            return addYears(timestamp, (int) Math.multiplyExact(numSteps, stepYears));
```

Comment:
Same consideration as for `MonthTimestampMicrosSampler`.

Reply:
fixed in https://github.com/questdb/questdb/pull/6517/commits/c9c81292ac6c7c9e4a5bd20c2a21603d51c03623